src/WinRT.Runtime/ComWrappersSupport.net5.cs

```diff
@@ -178,6 +178,10 @@ internal static IntPtr CreateCCWForObjectForABI(object obj, Guid iid)
                 Marshal.ThrowExceptionForHR(Marshal.QueryInterface(ccw, ref iid, out var iidCcw));
                 return iidCcw;
             }
+            catch (InvalidCastException ex)
```

Comment:
Could we perhaps do something like this instead?

```csharp
IntPtr ccw = ComWrappers.GetOrCreateComInterfaceForObject(obj, CreateComInterfaceFlags.TrackerSupport);

// Invoke 'IUnknown::QueryInterface' on the CCW
int hr = Marshal.QueryInterface(ccw, ref iid, out IntPtr ppv);

// Always release the original CCW after the call
_ = Marshal.Release(ccw);

if (hr < 0)
{
    [MethodImpl(MethodImplOptions.NoInlining)]
    static Exception GetException(object obj, Guid iid, int hr)
    {
        // Special case 'E_NOINTERFACE' to provide a better exception message
        if (hr == ExceptionHelpers.E_NOINTERFACE)
        {
            return new InvalidCastException($"Failed to create a CCW for object of type '{obj.GetType()}' for interface with IID '{iid.ToString().ToUpperInvariant()}': the specified cast is not valid.");
        }

        return Marshal.GetExceptionForHR(hr);
    }

    throw GetException(obj, iid, hr);
}

return ppv;
```

In particular this also avoids the try/catch/rethrow, which is a bit faster, and also directly uses `Marshal.Release`.